pages/api/session.js

```diff
@@ -44,40 +47,60 @@ async function login (req, res) {
       }
 
     const { session_token, user: {firstName, lastName, roles, capabilities, defaultEntityType, forcePasswordChange} } = await People.login(email, password);
-    req.session.session_token = session_token;
-    req.session.cache = {
-      firstName,
-      lastName,
-      roles,
-      capabilities,
-      defaultEntityType,
-      homepage: '/admin',
-      forcePasswordChange,
-    };
-
-    const token = generateToken({ firstName, lastName });
-    setCookie('token', token, { req, res, httpOnly: true, secure: true, sameSite: 'none' });
-
-    await req.session.save();    
-    res.status(OK).json({ forcePasswordChange });
-  
-  } catch (error) {
-    if (error instanceof UnauthorizedError) {
-      res.status(UNPROCESSABLE_ENTITY).json({message: "Invalid username or password."});
-      return;
-    } else {
-      await defaultErrorHandler(error, req, res);
+
+    const { origin, host } = req.headers;
+    const isFromCMS = origin.endsWith(host);
+
+    const cmsUsers = "CMS Users"
+    const isCMSUser = roles.includes(cmsUsers);
```

Comment:
Instead of checking roles, check capabilities instead.

```suggestion
    const canLogInToCMS = capabilities.includes('can_log_in_to_cms');
```